lib/models/ObjectLockConfiguration.ts

```diff
@@ -183,6 +185,8 @@ export default class ObjectLockConfiguration {
             this._config.rule = {};
             this._config.rule.mode = validMode.mode;
             this._config.rule[validTime.timeType!] = validTime.timeValue;
+            // Store the number of days
+            this._days = validTime.timeType === 'years' ? 365 * validTime.timeValue : validTime.timeValue;
```

Comment:
I don't know if that matters here, but I would expect a number of days to be different than 365 times a number of years, for example an object created on Dec 20th 2023 with a 2-year retention would have to be expired on 20th 2025 (I believe), which is one more day than 365*2 due to 2024 being a leap year.

Reply:
does this address this for you? 
feb89ea7ce740eb75b428e16c14b4ff1e9b3c44e